pkg/injector/pod_patch.go

```diff
@@ -142,10 +147,77 @@ func (i *injector) getPodPatchOperations(ar *v1beta1.AdmissionReview,
 			Value: value,
 		},
 	)
+	patchOps = append(patchOps, envPatchOps...)
 
 	return patchOps, nil
 }
 
+// This function add Dapr environment variables to all the containers in any Dapr enabled pod.
+// The containers can be injected or user defined.
+func addDaprEnvVarsToContainers(containers []corev1.Container) []PatchOperation {
+	portEnv := []corev1.EnvVar{
+		{
+			Name:  userContainerDaprHTTPPortName,
+			Value: strconv.Itoa(sidecarHTTPPort),
+		},
+		{
+			Name:  userContainerDaprGRPCPortName,
+			Value: strconv.Itoa(sidecarAPIGRPCPort),
+		},
+	}
+	envPatchOps := []PatchOperation{}
+	for i, container := range containers {
+		path := fmt.Sprintf("%s/%d/env", containersPath, i)
+		patchOps := getEnvPatchOperations(container.Env, portEnv, path)
+		if len(patchOps) != 0 {
+			envPatchOps = append(envPatchOps, patchOps...)
+		}
+	}
+	return envPatchOps
+}
+
+// This function only add new environment variables if they do not exist.
+// It does not override existing values for those variables if they have been defined already.
+func getEnvPatchOperations(envs []corev1.EnvVar, addEnv []corev1.EnvVar, path string) []PatchOperation {
+	if len(envs) == 0 {
+		// If there are no environment variables defined in the container, we initialize a slice of environment vars.
+		return []PatchOperation{
+			{
+				Op:    "add",
+				Path:  path,
+				Value: addEnv,
+			},
+		}
+	}
+	// If there are existing env vars, then we are adding to an existing slice of env vars.
+	path += "/-"
+
+	// Add only env vars that do not conflict with existing user defined/injected env vars.
+	toAdd := []corev1.EnvVar{}
+	for _, env := range addEnv {
+		hasEnv := false
+		for _, actual := range envs {
+			if actual.Name == env.Name {
+				hasEnv = true
+				break
+			}
+		}
+		if !hasEnv {
+			toAdd = append(toAdd, env)
+		}
+	}
+
+	patchOps := make([]PatchOperation, len(toAdd))
+	for i, value := range toAdd {
```

Comment:
We do not need this for-loop if we can use append EnvVar to patchOps at line 206.

Reply:
e.g.
```
        var patchOps []PatchOperation
	LoopEnv: for _, env := range addEnv {
		for _, actual := range envs {
			if actual.Name == env.Name {
				hasEnv = true
				goto LoopEnv
			}
		}
                patchOps[i] = PatchOperation{
			Op:    "add",
			Path:  path,
			Value: []corev1.EnvVar{value},
		}
	}
```

Reply:
does PathOperation accept the list of values? is it accepting only one value